src/rt-app_utils.c

```diff
@@ -321,3 +332,22 @@ void ftrace_write(int mark_fd, const char *fmt, ...)
 	}
 
 }
+
+int
+str2loglevel(char *s)
+{
+	for (int n=0; lls[n].val != LOG_LEVEL_END; n++) {
+		int equals = 1;
+		int done = 0;
+		for (char *hay = lls[n].s, *ndl = s; !done && equals; hay++, ndl++) {
+			equals = (toupper(*ndl) == toupper(*hay));
+			done = (! *ndl) && (! *hay);
+		}
```

Comment:
Why do you create your own string comparator instead of using strncmp ?

Reply:
because i need to compare the uppercase variant of the string(s). would you prefer it to copy the string while converting to uppercase first, then strcmp()ing them?

Reply:
1st of all do we really want to support both upper and lower case ? TBH i'm not sure
Then, if we really want to support both lower and upper case, why not adding directly both entries in the table ?

Reply:
1) i thought it should be 'debug', not 'DEBUG' purely because it reads nicer. but if you look at the help string, that already says 'DEBUG', so maybe that's fine?

2) yes.. could duplicate every entry and then convert during init, but that seems a bit much doesn't it? don't know of a way of doing that with the preprocessor, otherwise i'd have done it that way in the beginning. do you?

ALTERNATIVELY, and this is maybe the most straightforward solution: what about (LOG_LEVEL_debug) and no conversion needed? the interface in util.h are the macros (log_debug() etc) anyways...?

up to you...

Reply:
ok.